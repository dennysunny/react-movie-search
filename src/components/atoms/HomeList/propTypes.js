import { func, string, array } from 'prop-types';

export const propTypes = {
  title: string,
  list: array.isRequired,
  goToMovie: func.isRequired,
};

export const defaultProps = {
};
