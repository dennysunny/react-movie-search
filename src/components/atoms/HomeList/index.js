import React from 'react';

import { propTypes } from './propTypes';
import MediaImage from '../MediaImage';

const HomeList = ({ title, list, goToMovie }) => (
  <div className="home-list">
    <div className="home-list__title">{title}</div>
    <ul className="home-list__list">
      {list.map(show => (
        // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
        <li
          className="home-list__list-element"
          key={show.id}
          onClick={() => goToMovie(show.id)}
          onKeyDown={() => goToMovie(show.id)}
        >
          <MediaImage
            mediaType="poster"
            size={0}
            filePath={show.poster_path}
            name={show.title}
          />
          <div className="list-data">
            <div className="list-data__title">
              {`${show.title} (${show.vote_average})`}
            </div>
            <div className="list-data__genres">
              {show.genre_names.genre_name.map((genreName, i) => (
                show.genre_names.length === i + 1
                  ? genreName
                  : `${genreName}, `
              ))}
            </div>
          </div>
        </li>
      ))}
    </ul>
  </div>
);

HomeList.propTypes = propTypes;

export default HomeList;
