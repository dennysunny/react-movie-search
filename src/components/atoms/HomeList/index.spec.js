import React from 'react';
import { shallow } from 'enzyme';

import HomeList from '.';

const mockProps = {
  title: 'title',
  list: [
    {
      id: 1,
      poster_path: 'path/to/poster',
      title: 'title',
      vote_average: '1',
      genre_names: {
        genre_name: ['genre'],
      },
    },
  ],
  goToMovie: jest.fn(),
};

describe('Atoms/HomeList', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<HomeList {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle onClick event', () => {
    const wrapper = shallow(<HomeList {...mockProps} />);

    wrapper.find('.home-list__list-element').simulate('click');
    expect(mockProps.goToMovie).toHaveBeenCalled();

    wrapper.find('.home-list__list-element').simulate('keyDown');
    expect(mockProps.goToMovie).toHaveBeenCalled();
  });
});
