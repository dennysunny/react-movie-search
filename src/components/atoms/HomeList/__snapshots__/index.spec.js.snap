// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/HomeList should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeList
    list={
      Array [
        Object {
          "genre_names": Object {
            "genre_name": Array [
              "genre",
            ],
          },
          "id": 1,
          "poster_path": "path/to/poster",
          "title": "title",
          "vote_average": "1",
        },
      ]
    }
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="home-list__title"
        >
          title
        </div>,
        <ul
          className="home-list__list"
        >
          <li
            className="home-list__list-element"
          >
            <Link
              replace={false}
              to="/movie/1"
            >
              <Apollo(MediaImage)
                filePath="path/to/poster"
                mediaType="poster"
                name="title"
                size={0}
              />
              <div
                className="list-data"
              >
                <div
                  className="list-data__title"
                >
                  title (1)
                </div>
                <div
                  className="list-data__genres"
                >
                  genre, 
                </div>
              </div>
            </Link>
          </li>
        </ul>,
      ],
      "className": "home-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "home-list__title",
        },
        "ref": null,
        "rendered": "title",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="home-list__list-element"
            >
              <Link
                replace={false}
                to="/movie/1"
              >
                <Apollo(MediaImage)
                  filePath="path/to/poster"
                  mediaType="poster"
                  name="title"
                  size={0}
                />
                <div
                  className="list-data"
                >
                  <div
                    className="list-data__title"
                  >
                    title (1)
                  </div>
                  <div
                    className="list-data__genres"
                  >
                    genre, 
                  </div>
                </div>
              </Link>
            </li>,
          ],
          "className": "home-list__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/movie/1"
              >
                <Apollo(MediaImage)
                  filePath="path/to/poster"
                  mediaType="poster"
                  name="title"
                  size={0}
                />
                <div
                  className="list-data"
                >
                  <div
                    className="list-data__title"
                  >
                    title (1)
                  </div>
                  <div
                    className="list-data__genres"
                  >
                    genre, 
                  </div>
                </div>
              </Link>,
              "className": "home-list__list-element",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Apollo(MediaImage)
                    filePath="path/to/poster"
                    mediaType="poster"
                    name="title"
                    size={0}
                  />,
                  <div
                    className="list-data"
                  >
                    <div
                      className="list-data__title"
                    >
                      title (1)
                    </div>
                    <div
                      className="list-data__genres"
                    >
                      genre, 
                    </div>
                  </div>,
                ],
                "replace": false,
                "to": "/movie/1",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "filePath": "path/to/poster",
                    "mediaType": "poster",
                    "name": "title",
                    "size": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="list-data__title"
                      >
                        title (1)
                      </div>,
                      <div
                        className="list-data__genres"
                      >
                        genre, 
                      </div>,
                    ],
                    "className": "list-data",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "title (1)",
                        "className": "list-data__title",
                      },
                      "ref": null,
                      "rendered": "title (1)",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "genre, ",
                        ],
                        "className": "list-data__genres",
                      },
                      "ref": null,
                      "rendered": Array [
                        "genre, ",
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="home-list__title"
          >
            title
          </div>,
          <ul
            className="home-list__list"
          >
            <li
              className="home-list__list-element"
            >
              <Link
                replace={false}
                to="/movie/1"
              >
                <Apollo(MediaImage)
                  filePath="path/to/poster"
                  mediaType="poster"
                  name="title"
                  size={0}
                />
                <div
                  className="list-data"
                >
                  <div
                    className="list-data__title"
                  >
                    title (1)
                  </div>
                  <div
                    className="list-data__genres"
                  >
                    genre, 
                  </div>
                </div>
              </Link>
            </li>
          </ul>,
        ],
        "className": "home-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "home-list__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="home-list__list-element"
              >
                <Link
                  replace={false}
                  to="/movie/1"
                >
                  <Apollo(MediaImage)
                    filePath="path/to/poster"
                    mediaType="poster"
                    name="title"
                    size={0}
                  />
                  <div
                    className="list-data"
                  >
                    <div
                      className="list-data__title"
                    >
                      title (1)
                    </div>
                    <div
                      className="list-data__genres"
                    >
                      genre, 
                    </div>
                  </div>
                </Link>
              </li>,
            ],
            "className": "home-list__list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/movie/1"
                >
                  <Apollo(MediaImage)
                    filePath="path/to/poster"
                    mediaType="poster"
                    name="title"
                    size={0}
                  />
                  <div
                    className="list-data"
                  >
                    <div
                      className="list-data__title"
                    >
                      title (1)
                    </div>
                    <div
                      className="list-data__genres"
                    >
                      genre, 
                    </div>
                  </div>
                </Link>,
                "className": "home-list__list-element",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Apollo(MediaImage)
                      filePath="path/to/poster"
                      mediaType="poster"
                      name="title"
                      size={0}
                    />,
                    <div
                      className="list-data"
                    >
                      <div
                        className="list-data__title"
                      >
                        title (1)
                      </div>
                      <div
                        className="list-data__genres"
                      >
                        genre, 
                      </div>
                    </div>,
                  ],
                  "replace": false,
                  "to": "/movie/1",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "filePath": "path/to/poster",
                      "mediaType": "poster",
                      "name": "title",
                      "size": 0,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          className="list-data__title"
                        >
                          title (1)
                        </div>,
                        <div
                          className="list-data__genres"
                        >
                          genre, 
                        </div>,
                      ],
                      "className": "list-data",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "title (1)",
                          "className": "list-data__title",
                        },
                        "ref": null,
                        "rendered": "title (1)",
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "genre, ",
                          ],
                          "className": "list-data__genres",
                        },
                        "ref": null,
                        "rendered": Array [
                          "genre, ",
                        ],
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
